[Edit + Term Entry] React: Hooks (#1073)
Codecademy/docs: content/react/concepts/hooks/hooks.md

  - 'paths/front-end-engineer-career-path'
---

> In React, **hooks** are functions that give class-like abilities to function components, include [state](https://www.codecademy.com/resources/docs/react/state) and side-effects.

Reviewer: ```suggestion
In React, **hooks** are [functions](https://www.codecademy.com/resources/docs/javascript/functions) that give class-like abilities to function [components](https://www.codecademy.com/resources/docs/react/components), including [state](https://www.codecademy.com/resources/docs/react/state) and side-effects.
```

> There are a few rules when using hooks! Specifically:

Reviewer: ```suggestion
There are a few rules when using hooks, including the following:
```

Reviewer: Can we also rephrase the two rules on lines # 18 and # 19? Maybe something like the following:

> \- Hooks must be called from React functions (i.e. components or custom hooks) and not from a regular JavaScript function.
> \- Hooks should not be called inside of a loop, condition, or nested function. Instead, hooks should be called at the top level of a React function so they render in the same order every time.

> Hooks are imported at the top of a file from the `react` library:

Reviewer: ```suggestion
## Syntax
```

I would move this below the \```pseudo snippet, but above the **note**.

```pseudo
import React, { useHook } from 'react';
```

> > **Note:** The code snippet above is purely pseudocode and useHook is not an actual React hook.

Reviewer: ```suggestion
Hooks are imported at the top of a file from the `react` library.

> **Note:** The code snippet above is purely pseudocode and `useHook` is not an actual React hook. However, all hooks must begin with `use`.
```

> ```jsx

Reviewer: ```suggestion
```pseudo
```

> import React, { useState, useEffect } from 'react';

Reviewer: ```suggestion
import React, { useHook } from 'react';
```

Author: Hmm, do you think this may mislead readers into thinking there's an actual hook called `useHook`?

Reviewer: @krowter I see your point, but I also think its important to make sure this is in line with what a \```pseudo code block should look/be like in the "Syntax" section. How about we attach a **Note** that explains this point? Maybe something like the following?

> \> \*\*Note:\*\* The code snippet above is purely pseudocode and `useHook` is not an actual React hook.
